### ISSUE TITLE
defaultValue is mutated

### ISSUE THREAD

**Issue by AurelienRichez** [AlexGalays/immupdate#15]
While using deepUpdate I found out the hard way that the value provided as default was modified in place. I would guess it is not intended behaviour.

with immudate v 1.2.0 (imported via `space-lift`), I check and this is also the case with 1.2.3.

Here is a small script to reproduce : 

```
const { deepUpdate } = require('immupdate')

const valueToUpdate = { aField: ''}
const defaultValue = { foo: 'aDefaultValue' }
deepUpdate(valueToUpdate)
  .at('otherField')
  .withDefault(defaultValue)
  .at('foo')
  .set('valueChanged!')

// defaultValue.foo is now 'valueChanged!'
console.log(JSON.stringify(defaultValue))
```

**Comment by AurelienRichez** [comment]
I can try to look at it if and make a PR if it is not intended behaviour. Right now a good enough workaround is to just give a new object every time `{...defaultValue}`.

**Comment by AlexGalays** [comment]
This is most definitely a bug!

I just started adding support for updating `Option`s and it's quite an intrusive set of changes so the easiest would be for me to fix the above bug right after.

**Comment by AlexGalays** [comment]
Fixed in `1.3.0`!
